Java/core/src/main/java/com/amazon/randomcutforest/RandomCutForest.java

```diff
@@ -154,9 +168,21 @@
      */
     protected final boolean storeSequenceIndexesEnabled;
     /**
-     * Store the time information
+     * Enable compact representation
      */
     protected final boolean compactEnabled;
+    /**
+     * Enable float representation; the input/output is double[] but the internal
+     * representation would be float[]. There will be loss of precision in near
+     * neighbors because of the truncation. Otherwise, unless all numbers are in a
+     * very small range, the loss of precision is unlikely to have larger impact
+     * than changing the random seed.
+     */
+    protected final boolean singlePrecisionEnabled;
```

Comment:
We are currently proposing to support both single and double precision in 2.0. However, we're also planing to change this boolean flag to an enum value because think it would be clearer. https://github.com/aws/random-cut-forest-by-aws/issues/77